float top_roller
uint8 shoot_ball
